pkg/git/remote_git_resolver.go

```diff
@@ -59,19 +65,21 @@ func (*remoteGitResolver) Resolve(keychain GitKeychain, sourceConfig corev1alpha
 		}, nil
 	}
 
-	references, err := remote.Ls()
+	references, err := remote.List(&gogit.ListOptions{
+		Auth: auth,
+	})
 	if err != nil {
-		return corev1alpha1.ResolvedSourceConfig{}, errors.Wrap(err, "remote ls")
+		return corev1alpha1.ResolvedSourceConfig{}, errors.Wrap(err, "listing remote references")
 	}
 
-	for _, ref := range references {
-		for _, format := range refRevParseRules {
-			if fmt.Sprintf(format, sourceConfig.Git.Revision) == ref.Name {
+	for _, reference := range references {
+		for _, revRefParseRule := range refRevParseRules {
+			if fmt.Sprintf(revRefParseRule, sourceConfig.Git.Revision) == reference.Name().String() {
 				return corev1alpha1.ResolvedSourceConfig{
 					Git: &corev1alpha1.ResolvedGitSource{
 						URL:      sourceConfig.Git.URL,
-						Revision: ref.Id.String(),
-						Type:     sourceType(ref),
+						Revision: reference.Hash().String(),
+						Type:     referenceNameToType(reference.Name()),
```

Comment:
Compare to old implementation: https://github.com/pivotal/kpack/blob/d4de54b914365f6082bcbe938fa038aab18ed9b9/pkg/git/remote_git_resolver.go#L43